tmt-app/src/Components/MajorPage.jsx

```diff
@@ -1,10 +1,38 @@
-import React from "react";
+import { Container, Typography } from "@mui/material";
+import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+import { MAJORS_LIST } from "../constants/careerMajorList";
 
 function MajorPage() {
-  const majorChoice = useParams();
+  const params = useParams();
+  const [data, setData] = useState([]);
 
-  return <div>{majorChoice.major}</div>;
+  const searchData = (target) => {
+    for (let i = 0; i < MAJORS_LIST.length; i++) {
+      if (target === MAJORS_LIST[i].key) {
+        setData(MAJORS_LIST[i]);
+        console.log(MAJORS_LIST[i]);
+      }
+    }
+  };
+
+  useEffect(() => {
+    searchData(params.major);
+  }, [params]);
```

Comment:
Is there a reason we need an effect here? SearchData is currently just comparing and finding a value from a constant and doesn't feel like it warrants control over renders with this effect.

Can we have a better method just return the text we need directly into our JSX?

Reply:
After searching, it would be saved in a state variable so we don't have to search again. The effect would allow us to update the state variable when changes happen for the state.

Reply:
What reason would there be to need to save this to state?

Reply:
Yeah we honestly don't need state. We could save it in a variable since the data would not be changing. I will make the adjustment

Reply:
Yeah, MAJORS_LIST could also be a hash map, being able to access each major by key vs trying to check by index might save you a few lines in logic